src/PostgREST/CLI.hs

```diff
@@ -64,25 +68,33 @@ dumpSchema appState = do
       exitFailure
     Right sCache -> return $ JSON.encode sCache
 
+
 -- | Command line interface options
-data CLI = CLI
+data CLIOpts = CLIOpts
+  { example :: Example
+  , command :: Commands
+  }
+
+type Example = [Char]
+
+data Commands = Commands
   { cliCommand :: Command
   , cliPath    :: Maybe FilePath
   }
```

Comment:
The problem with adding ``CmdExample ExampleType`` to ``Command`` is that this is a value option. The other enums ``CmdDumpConfig`` and ``CmdDumpSchema`` are handled as flags. Combining them will complicate things in implementation as well. But maybe it is doable. I'll try this and get back to you.

Reply:
Also using this would make the ``interface help`` look like this:
```
Usage: postgrest-12.0.2 [-v|--version]
                        [--example EXAMPLETYPE | --dump-config | --dump-schema] [FILENAME]

```
instead of this more nice looking one:
```
Usage: postgrest-12.0.2 [-v|--version] [-e|--example EXAMPLETYPE]
                        [--dump-config | --dump-schema] [FILENAME]

```

Reply:
> Also using this would make the interface help look like this:
> 
> ```
> Usage: postgrest-12.0.2 [-v|--version]
>                         [--example EXAMPLETYPE | --dump-config | --dump-schema] [FILENAME]
> ```
> 

This means that `--example EXAMPLETYPE`, `--dump-config` and `--dump-schema` are mutually exclusive, which makes sense now. Before, you could do something like:

```sh
postgrest --example --dump-config
# or
postgrest --dump-config --example
```

And it will ignore the `--dump-config` and print the example directly. But now that `--example` has an argument, it complicates things as Wolfgang mentioned. So I think it's OK to have them mutually exclusive like that.

Reply:
@laurenceisla  I think the above is not possible to implement as it is inconsistent. Adding an argument option in `Command` that was previously just enum does not work.
As I have looked further into this, a pattern like:
```haskell
data Command
  = ...
  | CmdExample ExampleType
  | Something Argument
```
is used in subparsers. [ref](https://github.com/pcapriotti/optparse-applicative?tab=readme-ov-file#commands). Hence, implementing the above is inconsistent if not impossible. How about using subparsers to lift the interface something like:
```
$ postgrest dump [--dump-config | --dump-schema-cache]
$ postgrest example [--file | --db | --env]
```
I think it seems much simpler, extendible and easier to implement.

Reply:
@laurenceisla 
I have made the interface like you said earlier:
```
Usage: postgrest-12.0.2 [-v|--version]
                        [--example EXAMPLETYPE | --dump-config | --dump-schema] [FILENAME]

```
Please confirm and review this design so I can move to testing.

Reply:
Argh... this gets complicated due to the need of a parameter and default values for the `--example`. It's in conflict with the `CLI` type expected by the parser and generates the problems that both of you encountered along the way.

> How about using subparsers to lift the interface something like:

I think this may be the way to go in the future. It would need a rework of the CLI, I suppose.

> Please confirm and review this design so I can move to testing.

I'm sorry to keep going back and forth but what if, instead of using an option with value, we do the same as with `--dump-*`. I mean, to use: 

 `--example-file`, `--example-env`, `--example-db`. 

This would work as an extension of the already used `--example` and wouldn't change nor break the way the CLI is currently used. The drawback is that it's more verbose.

@taimoorzaeem @wolfgangwalther WDYT? I'm adding how the parsing would go in a review below.